### PR TITLE
Add CI formatting to lightning.qubit

### PR DESCRIPTION
**Pull request by mlxd**
**Context:** Standardisation of C++ styling across repository will help to ensure consistency across multiple developers. [ch7080]

**Description of the Change:** This PR adds a formatter script, and CI tests to run on PR creation. The formatter was run on the repository to provide an initial starting-point for future changes, hence the large line-number changes. In addition, some missing headers were identified and added.

**Benefits:** Standardisation of C++ styling will help ensure consistency across future developments of the project.

**Possible Drawbacks:** Additional efforts to format before creating a PR.

**Related GitHub Issues:** None.

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: Formatting check
 on:
-- pull_request
+  pull_request:
+    push:
+      branches:
+        - master
+      paths:
+        - "pennylane_lightning/**"
+        - "tests/**"
```
Comment (trbromley) [inline .github/workflows/format.yml:9]: Ah cool, I didn't know you could set it up to run only for certain changes!

File: bin/format [hunk 2]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python3
```
Comment (trbromley) [inline bin/format:1]: Is `bin/format` the usual spot to put this?

**Review by trbromley — APPROVED** [review]
Thanks @mlxd!

File: bin/format (hunk 2 above)
Comment (mlxd) [reply bin/format:1]: I think we can put this anywhere that seems sensible. In this case, since we run it as an executable script I added it here.

**Pull request merged**